src/internal.c

```diff
@@ -4196,8 +4196,20 @@ int DhAgree(WOLFSSL* ssl, DhKey* dhKey,
     else
 #endif
     {
-        ret = wc_DhAgree(dhKey, agree, agreeSz, priv, privSz, otherPub,
-                otherPubSz);
+#if !defined(HAVE_FIPS) && !defined(HAVE_SELFTEST)
+        ret = wc_DhCheckPubValue(ssl->buffers.serverDH_P.buffer,
+                    ssl->buffers.serverDH_P.length, otherPub, otherPubSz);
+        if (ret != 0) {
+    #ifdef OPENSSL_EXTRA
+            SendAlert(ssl, alert_fatal, illegal_parameter);
```

Comment:
Is there a reason to only send the alert in OpenSSL mode?

Reply:
This is a pattern I'm following with all the alert changes in non-TLS13 code.
The code currently doesn't send alerts in many cases.
Sending alerts is extra code size for little benefit and since we don't appear to be sending alerts as a rule, or only when critical, then I decided to only conditionally compile them in.
There was a recent PR from Tesfa as I remember where the alert was protected this way and I'm following this.

Let me know if you want me to change this pattern!